### PR TITLE
Namespace Refactor: Refactor map into namespace

### PR DESCRIPTION
**Pull request by simondevenish**
Current and future namespace refactors will format include directives as #include "directory/filename"

### CODE REVIEW

**Comment by simondevenish** [comment]
` 109 warnings and 1 error generated.
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.`
@HexDecimal can you retrigger the job?

File: src/map/dungeon.hpp [hunk 1]
```diff
@@ -30,24 +30,25 @@
...
-#include "map_cell.hpp"
+#include "map/cell.hpp"
 #include "mob_creature.hpp"
 #include "util_cavegen.hpp"
 #include "util_cellular.hpp"
 #include "util_clouds.hpp"
 
 class Player;
-class LightMap;
+class map::LightMap;
```
Comment (HexDecimal) [inline src/map/dungeon.hpp]: Rerunning the job would trigger the 1 error again:
```
/Users/runner/work/treeburner/treeburner/src/map/dungeon.hpp:40:7: error: forward declaration of class cannot have a nested name specifier
class map::LightMap;
```
Could probably be fixed like this?  I'm not sure.
```suggestion
namespace map {
class LightMap;
}  // namespace map
```
Ideally these forward references would be minimized, but that will likely be difficult.

Jobs are automatically rerun when you push another commit.

Comment (simondevenish) [reply src/map/dungeon.hpp]: Ah my bad. I missed this. This is generally how I have been fixing these forward references

**Comment by HexDecimal** [comment]
Are actions not enabled for your repo? https://github.com/odiminox/treeburner/actions

I'm not sure but I think these are disabled by default until you turn them on.  I know I can force them to run by closing and reopening the PR.

**Comment by simondevenish** [comment]
I've turned them on, you are right, it was disabled by default. Feel free to close/reopen this PR to retrigger for now and in the future it will be working on my end

**Review by HexDecimal — APPROVED** [review]

**Pull request merged**